### ISSUE TITLE
nc-datadir not updating apache logs

### ISSUE THREAD

**Issue by pilepalle** [nextcloud/nextcloudpi#507]
Hi guys, 
I am new in the Nextcloudpi environment. I had the NextcloudPI image ready to go. Then I thought I'm going to update to the latest versions. Unfortunately, something went wrong. Since then, I'm no longer on the Nextcloud. It can be seen in the ncp-report that the httpd service is down and the port check 80 and 443 are closed. Does anyone have an idea what that can be?

pi@nextcloudplus:/media $ sudo ncp-report
<--! Paste this in GitHub report -->

<details>
<summary>NextCloudPlus diagnostics</summary>

```

NextCloudPlus version  v0.56.4
NextCloudPlus image    NextCloudPlus_04-21-18
distribution           Raspbian GNU/Linux 9 \n \l
automount              yes
USB devices            sda sdb 
datadir                /media/USBdrive/ncdata
data in SD             no
data filesystem        btrfs
data disk usage        124G/1.4T
rootfs usage           2.3G/15G
swapfile               /var/swap
Nextcloud check        ok
Nextcloud version      13.0.2.1
HTTPD service          down
PHP service            up
MariaDB service        up
Redis service          up
Postfix service        up
internet check         ok
port check 80          closed
port check 443         closed
IP                     192.168.176.200
gateway                192.168.176.2
interface              eth0
certificates           ekkr.net
NAT loopback           yes
uptime                 4min

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "localhost",
            "5": "nextcloudplus.local",
            "1": "192.168.176.200",
            "4": "ekkr.net",
            "2": "ekkr.net"
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/ekkr.net",
        "dbtype": "mysql",
        "version": "13.0.2.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "overwriteprotocol": "https",
        "maintenance": false,
        "logfile": "\/media\/USBdrive\/ncdata\/nextcloud.log",
        "loglevel": "2",
        "log_type": "file",
        "mail_smtpsecure": "ssl",
        "mail_smtpport": "465",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
    }
}

```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Fri May 18 23:08:41.620692 2018] [authz_host:error] [pid 948:tid 1851356208] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/poweroff.svg failed, reason: unable to get the remote host name
[Fri May 18 23:08:41.620947 2018] [authz_host:error] [pid 948:tid 1817801776] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/reboot.svg failed, reason: unable to get the remote host name
[Fri May 18 23:08:41.698289 2018] [authz_host:error] [pid 948:tid 1809413168] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Fri May 18 23:08:41.700604 2018] [authz_host:error] [pid 948:tid 1801024560] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/menu.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.713182 2018] [authz_host:error] [pid 948:tid 1884910640] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/info-white.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.713182 2018] [authz_host:error] [pid 948:tid 1926853680] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/wizard.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.713617 2018] [authz_host:error] [pid 948:tid 1884910640] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/dashboard.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.713618 2018] [authz_host:error] [pid 948:tid 1901687856] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/settings-white.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.713662 2018] [authz_host:error] [pid 948:tid 1859744816] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /img/nc-button.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.715297 2018] [authz_host:error] [pid 948:tid 1926853680] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /core/img/actions/triangle-s.svg failed, reason: unable to get the remote host name, referer: https://192.168.176.200:4443/ncp.css
[Fri May 18 23:08:41.760422 2018] [authz_host:error] [pid 948:tid 1868133424] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /ncp-output.php failed, reason: unable to get the remote host name
[Fri May 18 23:08:42.365057 2018] [proxy_fcgi:error] [pid 948:tid 1868133424] [client 192.168.179.218:59324] AH01071: Got error 'PHP message: PHP Warning:  touch(): Utime failed: Permission denied in /var/www/ncp-web/ncp-output.php on line 81\n'
[Fri May 18 23:08:47.278730 2018] [proxy_fcgi:error] [pid 948:tid 1809413168] [client 192.168.179.218:59324] AH01071: Got error 'PHP message: PHP Warning:  fclose(): supplied resource is not a valid stream resource in /var/www/ncp-web/L10N.php on line 163\n'
[Fri May 18 23:08:47.293862 2018] [authz_host:error] [pid 948:tid 1918465072] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Fri May 18 23:08:49.037050 2018] [proxy_fcgi:error] [pid 948:tid 1918465072] [client 192.168.179.218:59324] AH01071: Got error 'PHP message: PHP Warning:  fclose(): supplied resource is not a valid stream resource in /var/www/ncp-web/L10N.php on line 163\n'
[Fri May 18 23:09:22.376133 2018] [authz_host:error] [pid 948:tid 1893299248] [client 192.168.179.218:59324] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Fri May 18 23:09:22.409337 2018] [proxy_fcgi:error] [pid 948:tid 1893299248] [client 192.168.179.218:59324] AH01071: Got error 'PHP message: PHP Warning:  fclose(): supplied resource is not a valid stream resource in /var/www/ncp-web/L10N.php on line 163\n'
[Fri May 18 23:09:24.612184 2018] [proxy_fcgi:error] [pid 948:tid 1868133424] [client 192.168.179.218:59324] AH01067: Failed to read FastCGI header
[Fri May 18 23:09:24.612348 2018] [proxy_fcgi:error] [pid 948:tid 1868133424] (104)Connection reset by peer: [client 192.168.179.218:59324] AH01075: Error dispatching request to :4443: 
[Fri May 18 23:09:25.766200 2018] [mpm_event:notice] [pid 934:tid 1992581120] AH00491: caught SIGTERM, shutting down

```
</details>

<details>
<summary>Database logs</summary>

```

2018-05-19 13:28:36 1639170880 [Note] /usr/sbin/mysqld: Shutdown complete

2018-05-19 13:29:10 1988550656 [Note] InnoDB: Using mutexes to ref count buffer pool pages
2018-05-19 13:29:10 1988550656 [Note] InnoDB: The InnoDB memory heap is disabled
2018-05-19 13:29:10 1988550656 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2018-05-19 13:29:10 1988550656 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2018-05-19 13:29:10 1988550656 [Note] InnoDB: Compressed tables use zlib 1.2.8
2018-05-19 13:29:10 1988550656 [Note] InnoDB: Using Linux native AIO
2018-05-19 13:29:10 1988550656 [Note] InnoDB: Using generic crc32 instructions
2018-05-19 13:29:10 1988550656 [Note] InnoDB: Initializing buffer pool, size = 128.0M
2018-05-19 13:29:11 1988550656 [Note] InnoDB: Completed initialization of buffer pool
2018-05-19 13:29:11 1988550656 [Note] InnoDB: Highest supported file format is Barracuda.
2018-05-19 13:29:11 1988550656 [Note] InnoDB: 128 rollback segment(s) are active.
2018-05-19 13:29:11 1988550656 [Note] InnoDB: Waiting for purge to start
2018-05-19 13:29:12 1988550656 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.35-80.0 started; log sequence number 430947018
2018-05-19 13:29:12 1442837312 [Note] InnoDB: Dumping buffer pool(s) not yet started
2018-05-19 13:29:12 1988550656 [Note] Plugin 'FEEDBACK' is disabled.
2018-05-19 13:29:12 1988550656 [Note] Server socket created on IP: '127.0.0.1'.
2018-05-19 13:29:12 1988550656 [Note] /usr/sbin/mysqld: ready for connections.
Version: '10.1.23-MariaDB-9+deb9u1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  Raspbian 9.0

```
</details>

<details>
<summary>Nextcloud logs</summary>

```

{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:11+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Repair::step: Repair step: Rename folder with user specific keys","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:11+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Repair::step: Repair step: Add log rotate job","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:11+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Repair::step: Repair step: Clear frontend caches","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:11+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Repair::info: Repair info: Image cache cleared","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:16+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Repair::info: Repair info: SCSS cache cleared","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:19+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Repair::info: Repair info: JS cache cleared","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:19+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Updater::startCheckCodeIntegrity: Starting code integrity check...","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:39+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Updater::finishedCheckCodeIntegrity: Finished code integrity check","userAgent":"--","version":"13.0.1.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:40+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Updater::updateEnd: Update successful","userAgent":"--","version":"13.0.2.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:40+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Updater::maintenanceDisabled: Turned off maintenance mode","userAgent":"--","version":"13.0.2.1"}
{"reqId":"yk00HnjWlovtL7vkYJ2M","level":1,"time":"2018-05-19T10:35:40+00:00","remoteAddr":"","user":"--","app":"updater","method":"--","url":"--","message":"\\OC\\Updater::resetLogLevel: Reset log level to Warning(2)","userAgent":"--","version":"13.0.2.1"}
{"reqId":"1TXLSSOxUTAKeLf3LiqY","level":2,"time":"2018-05-19T11:19:19+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"ug9L1XG5PN7taS2mooSU","level":2,"time":"2018-05-19T11:19:21+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"pKXAPgpi7FKUBck3tcFj","level":2,"time":"2018-05-19T11:19:32+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"aSZTvC5jysZvRJvxKZ2f","level":2,"time":"2018-05-19T11:25:41+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"vAgKCH46GPnEYPwTGMJO","level":2,"time":"2018-05-19T11:29:27+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"8Y0fwmiY6pdXihRrQKYT","level":2,"time":"2018-05-19T11:30:12+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"T5dAXDVO8MZT2QlQg8gy","level":2,"time":"2018-05-19T11:30:22+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"NFnrMLoasKnv6PxGWdGl","level":2,"time":"2018-05-19T11:33:06+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}
{"reqId":"49wCVCHnGzvE16p3W6vR","level":2,"time":"2018-05-19T11:33:11+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"--","url":"--","message":"mood needs circles","userAgent":"--","version":"13.0.2.1"}

```
</details>

pi@nextcloudplus:/media $ 

Thanks

**Comment by nachoparker** [comment]
hi,

what's the output of 

```
systemctl status apache2
```

**Comment by pilepalle** [comment]
● apache2.service - The Apache HTTP Server
   Loaded: loaded (/lib/systemd/system/apache2.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sat 2018-05-19 14:03:48 CEST; 3h 5min ago
  Process: 3227 ExecStart=/usr/sbin/apachectl start (code=exited, status=1/FAILURE)

May 19 14:03:48 nextcloudplus systemd[1]: Starting The Apache HTTP Server...
May 19 14:03:48 nextcloudplus apachectl[3227]: (2)No such file or directory: AH02291: Cannot access directory '/var/www/nextcloud/data/' for error log of vhost defined at /etc/apache2/sites-enabled/nextcloud.conf:2
May 19 14:03:48 nextcloudplus apachectl[3227]: AH00014: Configuration check failed
May 19 14:03:48 nextcloudplus apachectl[3227]: Action 'start' failed.
May 19 14:03:48 nextcloudplus apachectl[3227]: The Apache error log may have more information.
May 19 14:03:48 nextcloudplus systemd[1]: apache2.service: Control process exited, code=exited status=1
May 19 14:03:48 nextcloudplus systemd[1]: Failed to start The Apache HTTP Server.
May 19 14:03:48 nextcloudplus systemd[1]: apache2.service: Unit entered failed state.
May 19 14:03:48 nextcloudplus systemd[1]: apache2.service: Failed with result 'exit-code'.
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~

**Comment by pilepalle** [comment]
I've just set up the nextcloudpi on another raspi. The update to 0.56.4 and the Nextcloud from 13.0.1.1 to 13.02.1 performed. After a reboot same problem.

**Comment by nachoparker** [comment]
how did you perform the Nextcloud update? from NC itself, or from ncp-web?

**Comment by nachoparker** [comment]
ok, that feature was just added recently. If you could help me fix this... can I get the output of the process?

You can get it from `/var/log/ncp.log`, or from ncp-web/ncp-config right after you launch `nc-update-nextcloud`

**Comment by nachoparker** [comment]
you don't need to issue those commands. It should get fixed by updating to `v0.56.6`

thanks!

**Comment by nachoparker** [comment]
either your pi is really busy doing something right now, or you have hardware issues (bad powersupply/SD card)

you have tried rebooting right?
